### ISSUE TITLE
Doc: Review and Reorder Navigation

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#16719]
The documentation needs to be set up so:

* The navigation on the left sidebar is relatively easy to browse.
    * The browsing needs to be for people who don't already know dotCMS, so it can help people reviewing and evaluate the product find the appropriate docs, and quickly see what topics are covered.
* The navigation on the left sidebar is context-sensitive.
    * This means that the titles shown in the nav sidebar should be short, taking into account the context-sensitivity of them being within topics that are already shown.
* The document titles are verbose.
    * The document titles are what's displayed in the search results, so they need to be verbose enough that someone seeing the search result can tell the full context of the doc from just the title.
* Appropriate sections are automatically expanded so that:
    * Topics people are likely to be looking for are visible without them having to click too many times, but...
    * The docs are still within logical groupings (with headers & sub-headers) to make it easier to find specific documents without having to look through an un-grouped list of many documents.
        * The ideal max number of docs in each indented section (beneath any header or sub-header) is 7.
        * The absolute max number in any section should be 12.
* The docs in each section are ordered to make sense - **especially to users who don't already know the software**.
    * In other words, we don't need to bend over backward to help a 10-year dotCMS veteran find things.
        * They'll find them, and they'll already have bookmarks to the docs they user frequently.
    * Instead we need to optimize it for new users and people evaluating the product.

This project is in progress, but needs additional review and re-working.

## Consider Adding:

* Ability to highlight documents in the nav tree in some way so they stand out, especially to new users.
    * For example, the "What to Know Before Push Publishing" section should be first in the list (in the section it's in), but it should **also** ideally be highlighted in some way.
